### PR TITLE
feat: global default parameters

### LINKED ISSUE #129: Add default params configuration
**Feature Request:**

It would be awesome to be able to automatically pass a set of default parameters with each imgix request. For example, I'd like to have `auto=format,compress` and `fit=crop` sent with every request and not have to specify them within every `ix-params` object.

Perhaps something like `imgix.config.defaultParams = object` could work well for this?

### PR DESCRIPTION
**Pull request by frederickfogerty**
## Description

This PR allows developers to set default parameters for every img tag. This prevents them from having to set the same parameters over and over again.

This fixes #129 

## Steps to test

Review new unit tests.

### CODE REVIEW

File: src/ImgixTag.js [hunk 1]
```diff
@@ -49,10 +49,14 @@ var ImgixTag = (function() {
   }
 
   ImgixTag.prototype._extractBaseParams = function() {
-    var params;
+		var params = {};
+
+		if (this.settings.defaultParams) {
```
Comment (jayeb) [inline src/ImgixTag.js]: Should we type-check this to make sure it's an object?

File: src/ImgixTag.js [hunk 2]
```diff
@@ -49,10 +49,14 @@ var ImgixTag = (function() {
   }
 
   ImgixTag.prototype._extractBaseParams = function() {
-    var params;
+		var params = {};
```
Comment (jayeb) [inline src/ImgixTag.js]: [nit] Spaces and tabs, my friend... spaces and tabs.

**Comment by frederickfogerty** [comment]
1. I've added two tests for this in `spec/ImgixTagSpec.js` - are these sufficient? 

2. I think this is a good idea - I'll update this PR once I've implemented this.

File: src/ImgixTag.js (hunk 1 above)
Comment (frederickfogerty) [reply src/ImgixTag.js]: Yep, I can do this

File: src/ImgixTag.js (hunk 2 above)
Comment (frederickfogerty) [reply src/ImgixTag.js]: 🙈 https://www.youtube.com/watch?v=cowtgmZuai0

**Comment by frederickfogerty** [comment]
@jayeb (1) was discussed offline, and I believe I've implemented (2). Could you give this another review?

File: src/ImgixTag.js [hunk 3]
```diff
@@ -81,7 +83,7 @@ var ImgixTag = (function() {
 
     if (this.settings.includeLibraryParam) {
       params.ixlib = 'imgixjs-' + imgix.VERSION;
-    }
+		}
```
Comment (jayeb) [inline src/ImgixTag.js]: [nit] Tabs & spaces, again. Is this perchance a Prettify thing that's breaking down?

File: src/defaultConfig.js [hunk 4]
```diff
@@ -2,7 +2,8 @@ module.exports = {
   // URL assembly
   host: null,
   useHttps: true,
-  includeLibraryParam: true,
+	includeLibraryParam: true,
```
Comment (jayeb) [inline src/defaultConfig.js]: [nit] T&S

File: src/ImgixTag.js (hunk 3 above)
Comment (frederickfogerty) [reply src/ImgixTag.js]: Gah! I think it's time to tame the beast and work out what's going on here.

**Comment by frederickfogerty** [comment]
@jayeb those pesky space and tab issues should be gone for good now!

**Pull request merged**